In [25]:
import spacy
nlp = spacy.load('en')
from spacy.lang.en import English
#parser = English()
#nlp = spacy.load('en_core_web_sm')
example='''
That this House notes the announcement of 300 redundancies at the Nestlé manufacturing factories in York, Fawdon, Halifax and Girvan and that production of the Blue Riband bar will be transferred to Poland; acknowledges in the first three months of 2017 Nestlé achieved £21 billion in sales, a 0.4 per cent increase over the same period in 2016; further notes 156 of these job losses will be in York, a city that in the last six months has seen 2,000 job losses announced and has become the most inequitable city outside of the South East, and a further 110 jobs from Fawdon, Newcastle; recognises the losses come within a month of triggering Article 50, and as negotiations with the EU on the UK leaving the EU and the UK's future with the EU are commencing; further recognises the cost of importing products, including sugar, cocoa and production machinery, has risen due to the weakness of the pound and the uncertainty over the UK's future relationship with the single market and customs union; and calls on the Government to intervene and work with hon. Members, trades unions GMB and Unite and the company to avert these job losses now and prevent further job losses across Nestlé.'''

In [26]:
def entities(example, show=True):
    if show: print(example)
    #parsedEx = parser(example)
 
    print("-------------- entities only ---------------")
    # if you just want the entities and nothing else, you can do access the parsed examples "ents" property like this:
    doc = nlp(example)
    ents=doc.ents
    print(ents)
    tags={}
    for entity in ents:
        #print(entity.label, entity.label_, ' '.join(t.orth_ for t in entity))
        term=' '.join(t.orth_ for t in entity)
        if ' '.join(term) not in tags:
            tags[term]=[(entity.label, entity.label_)]
        else:
            tags[term].append((entity.label, entity.label_))
    print(tags)

In [27]:
entities(example)


That this House notes the announcement of 300 redundancies at the Nestlé manufacturing factories in York, Fawdon, Halifax and Girvan and that production of the Blue Riband bar will be transferred to Poland; acknowledges in the first three months of 2017 Nestlé achieved £21 billion in sales, a 0.4 per cent increase over the same period in 2016; further notes 156 of these job losses will be in York, a city that in the last six months has seen 2,000 job losses announced and has become the most inequitable city outside of the South East, and a further 110 jobs from Fawdon, Newcastle; recognises the losses come within a month of triggering Article 50, and as negotiations with the EU on the UK leaving the EU and the UK's future with the EU are commencing; further recognises the cost of importing products, including sugar, cocoa and production machinery, has risen due to the weakness of the pound and the uncertainty over the UK's future relationship with the single market and customs union; 

In [28]:
q= "Bob Smith was in the Houses of Parliament the other day"
entities(q)

Bob Smith was in the Houses of Parliament the other day
-------------- entities only ---------------
(Bob Smith, the Houses of Parliament, the other day)
{'Bob Smith': [(378, 'PERSON')], 'the Houses of Parliament': [(381, 'ORG')], 'the other day': [(388, 'DATE')]}


In [ ]:
k = "Tree is near the house"

In [ ]:
entities(k)

<H2>Stanford NER</H2>

In [1]:
import os
java_path = "C:/Program Files/Java/jdk1.8.0_151/bin"
os.environ['JAVA_HOME'] = java_path
#import nltk
#nltk.internals.config_java("C:/Program Files/Java/jdk1.6.0_30/bin/java.exe")

In [24]:
from nltk.tag import StanfordNERTagger
from nltk.tokenize import word_tokenize

st = StanfordNERTagger('E:/stanford-ner-2018-02-27/classifiers/english.all.3class.distsim.crf.ser.gz',
  'E:/stanford-ner-2018-02-27/stanford-ner.jar',encoding='utf-8')

text = 'While in France, Christine Lagarde discussed short-term stimulus efforts in a recent interview with the Wall Street Journal.'
k = "Tree is near the house"

tokenized_text = word_tokenize(k)
classified_text = st.tag(tokenized_text)

print(classified_text)

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)


[('Tree', 'O'), ('is', 'O'), ('near', 'O'), ('the', 'O'), ('house', 'O')]


<H2>Let's Code</H2>

<H2>Get all category names in quick draw dataset</H2>

In [2]:
category_file = open("categories.txt", "r")


In [3]:
category_lis = category_file.read().split('\n')

In [4]:
category_lis

['aircraft carrier',
 'airplane',
 'alarm clock',
 'ambulance',
 'angel',
 'animal migration',
 'ant',
 'anvil',
 'apple',
 'arm',
 'asparagus',
 'axe',
 'backpack',
 'banana',
 'bandage',
 'barn',
 'baseball',
 'baseball bat',
 'basket',
 'basketball',
 'bat',
 'bathtub',
 'beach',
 'bear',
 'beard',
 'bed',
 'bee',
 'belt',
 'bench',
 'bicycle',
 'binoculars',
 'bird',
 'birthday cake',
 'blackberry',
 'blueberry',
 'book',
 'boomerang',
 'bottlecap',
 'bowtie',
 'bracelet',
 'brain',
 'bread',
 'bridge',
 'broccoli',
 'broom',
 'bucket',
 'bulldozer',
 'bus',
 'bush',
 'butterfly',
 'cactus',
 'cake',
 'calculator',
 'calendar',
 'camel',
 'camera',
 'camouflage',
 'campfire',
 'candle',
 'cannon',
 'canoe',
 'car',
 'carrot',
 'castle',
 'cat',
 'ceiling fan',
 'cello',
 'cell phone',
 'chair',
 'chandelier',
 'church',
 'circle',
 'clarinet',
 'clock',
 'cloud',
 'coffee cup',
 'compass',
 'computer',
 'cookie',
 'cooler',
 'couch',
 'cow',
 'crab',
 'crayon',
 'crocodile',
 'crow

In [5]:
text = "WoW!!!! I love cats and dogs."

In [6]:
from nltk.tokenize import word_tokenize
from nltk.corpus import stopwords
import string
def preprocess_text(text):
    tokens = word_tokenize(text)
    # convert to lower case
    tokens = [w.lower() for w in tokens]
    # remove punctuation from each word

    table = str.maketrans('', '', string.punctuation)
    stripped = [w.translate(table) for w in tokens]
    # remove remaining tokens that are not alphabetic
    words = [word for word in stripped if word.isalpha()]
    # filter out stop words

    stop_words = set(stopwords.words('english'))
    words = [w for w in words if not w in stop_words]
    return words


In [7]:
lis=preprocess_text(text)

In [9]:
lis

['wow', 'love', 'cats', 'dogs']

In [10]:
from fuzzywuzzy import fuzz
from fuzzywuzzy import process
process.extractOne('near', category_lis)


('ear', 86)

In [99]:
import nltk
from nltk.tag import StanfordNERTagger
from nltk.tokenize import word_tokenize
import sys
import warnings

warnings.simplefilter('always', DeprecationWarning)

def extract_entities(text):
    
    st = StanfordNERTagger('E:/stanford-ner-2018-02-27/classifiers/english.all.3class.distsim.crf.ser.gz',
                           'E:/stanford-ner-2018-02-27/stanford-ner.jar',encoding='utf-8')
    #tokenized_text = word_tokenize(text)
    
    classified_text = st.tag(text)
    return classified_text

In [100]:
sample="cats and dogs, I love them"

In [101]:
sample=preprocess_text(sample)

In [102]:
warnings.simplefilter('always', DeprecationWarning)
extracted_lis=extract_entities(sample)

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)


In [103]:
extracted_lis

[('cats', 'O'), ('dogs', 'O'), ('love', 'O')]

<H2>Load google pretrained word2vec model to match sementic similarity of the category list and the text descriptions</H2>

In [16]:
import gensim

# Load Google's pre-trained Word2Vec model.
model = gensim.models.KeyedVectors.load_word2vec_format('./GoogleNews-vectors-negative300.bin', binary=True) 

In [20]:
model.similarity('trees','tree')

0.8293122426257508

In [110]:
import difflib
import sys
import warnings

if not sys.warnoptions:
    warnings.simplefilter("ignore")
def match_entities_with_dataset(text,category_lis):
    word_lis=preprocess_text(text)
    word_lis=extract_entities(word_lis)
    temp=[]
    for k in word_lis:
      t = difflib.get_close_matches(k[0], category_lis,1)
      if (t):
            if((t[0] in model.vocab) and (k[0] in model.vocab)and(model.similarity(k[0],t[0])>0.80)):
                
                temp.append(t[0])
    return temp

In [111]:
match_entities_with_dataset('trees near my House',category_lis)

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)


['tree', 'house']

<H2> Let us use mcTest dataset to get the narrative text description</H2>

In [23]:
import pandas as pd
#df=pd.read_csv('mc500_train.csv', sep=',',header=None)

In [24]:
df=pd.read_csv('mc500_train.csv', sep=',',header=None)

In [25]:
df.head()

0                                                  1   \
0  mc500.train.0  Author: 2778399758;Work Time(s): 839;Qual. sco...   
1  mc500.train.1  Author: 3208685851;Work Time(s): 836;Qual. sco...   
2  mc500.train.2  Author: 3165469218;Work Time(s): 1546;Qual. sc...   
3  mc500.train.3  Author: 3323630398;Work Time(s): 712;Qual. sco...   
4  mc500.train.4  Author: 646501657;Work Time(s): 743;Qual. scor...   

                                                  2   \
0  Alyssa got to the beach after a long trip. She...   
1  One morning, Elena woke up, much like she did ...   
2  Billy was like a king on the school yard. A ki...   
3  Billy went to the farm to buy some beef for hi...   
4  The road to Grandpa's house was long and windi...   

                                                  3   \
0                       one: What city is Alyssa in?   
1  multiple: What is the very first thing Elena d...   
2                      multiple: Billy was like a...   
3               one: What did Billy buy at the farm?   
4         multiple: Why did Grandpa answer the door?   

                           4                                   5   \
0                        trip                               Miami   
1  she says hello to the tree  she throws the covers on the floor   
2                      prince                               dummy   
3                        Beef                             Chicken   
4  Because he saw the insects           Because Jimmy was walking   

                          6                                  7   \
0                    Atlanta                              beach   
1  she says hello to the sun                she gets out of bed   
2                      queen                               king   
3                       Cows                              Fence   
4      Because Jimmy knocked  Because the trip took a long time   

                                                  8                9   \
0                   one: Why did Alyssa go to Miami?             swim   
1  one: At what time of day does this story take ...  Before the tree   
2                           multiple: Who was Billy?   The skinny kid   
3   multiple: What color were the spots on the cows?             Blue   
4           one: Where do Jimmy and his Grandpa sit?       On insects   

             ...              \
0            ...               
1            ...               
2            ...               
3            ...               
4            ...               

                                                  13  \
0       multiple: How many friends does Alyssa have?   
1    multiple: What happened to Mr. Fish in the end?   
2          one: What did the boy grab from his home?   
3  multiple: How many chickens were on the purple...   
4  multiple: What was Jimmy's favorite insect in ...   

                                             14  \
0                                             1   
1  He got put back in the bowl, but he was dead   
2                                        a rock   
3                                           Ten   
4                                        Beetle   

                                           15                        16  \
0                                           2                         3   
1  He started swimming around in Elena's hand  He got put in a new bowl   
2                                     a bench                    a tree   
3                                         Six                      Five   
4                                    Lady bug                      Moth   

                                                  17  \
0                                                  4   
1  He got put back in his bowl and started swimmi...   
2                                       Fishing line   
3                                               Four   
4                                      Lightning bug   

                               

In [114]:
import warnings

with warnings.catch_warnings():
    warnings.filterwarnings("ignore",category=DeprecationWarning)
    from nltk.tag import StanfordNERTagger 
result=[]
for txt in df[2]:
    result.extend(match_entities_with_dataset(txt,category_lis))

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\appdata\local\programs\python\python36\lib\site-packages\nltk\tag\stanford.py:183: DeprecationWarning: 
The StanfordTokenizer will be deprecated in version 3.2.5.
Please use nltk.tag.corenlp.CoreNLPPOSTagger or nltk.tag.corenlp.CoreNLPNERTagger instead.
  super(StanfordNERTagger, self).__init__(*args, **kwargs)
c:\users\admin\a

In [115]:
##Final result after namming the named entities to category
result

['beach',
 'beach',
 'house',
 'house',
 'steak',
 'house',
 'bed',
 'tree',
 'tree',
 'fish',
 'fish',
 'fish',
 'castle',
 'fish',
 'fish',
 'face',
 'bench',
 'tree',
 'line',
 'line',
 'tree',
 'bench',
 'bench',
 'face',
 'line',
 'face',
 'saw',
 'saw',
 'fence',
 'fence',
 'house',
 'tree',
 'house',
 'house',
 'lightning',
 'house',
 'door',
 'lightning',
 'house',
 'house',
 'tree',
 'television',
 'television',
 'telephone',
 'lightning',
 'rain',
 'potato',
 'bed',
 'strawberry',
 'spider',
 'saw',
 'spider',
 'spider',
 'spider',
 'spider',
 'spider',
 'saw',
 'horse',
 'cow',
 'horse',
 'cow',
 'dog',
 'sandwich',
 'fish',
 'sandwich',
 'horse',
 'cow',
 'onion',
 'potato',
 'horse',
 'cow',
 'dog',
 'sandwich',
 'fish',
 'sandwich',
 'house',
 'face',
 'saw',
 'saw',
 'dog',
 'dog',
 'bicycle',
 'bicycle',
 'river',
 'house',
 'bicycle',
 'river',
 'river',
 'river',
 'tree',
 'house',
 'ladder',
 'tree',
 'house',
 'ladder',
 'bucket',
 'bucket',
 'ladder',
 'ladder',
 '

In [117]:
##Eliminating duplicates to check the number of categories identified
result_final = list(set(result))

In [118]:
len(result_final)

163

163 categories are detected. We can train the NER and word2vec model to improve this detection rate. For instance the categories like "school bus" and others that have two consecutive words are not detected by our model. This can be improved by training.